chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/impl/FilterDAOImpl.java

```diff
@@ -0,0 +1,158 @@
+package gov.healthit.chpl.dao.impl;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import javax.persistence.Query;
+
+import org.springframework.context.i18n.LocaleContextHolder;
+import org.springframework.context.support.DefaultMessageSourceResolvable;
+import org.springframework.stereotype.Repository;
+
+import gov.healthit.chpl.auth.Util;
+import gov.healthit.chpl.auth.entity.UserEntity;
+import gov.healthit.chpl.dao.FilterDAO;
+import gov.healthit.chpl.dto.FilterDTO;
+import gov.healthit.chpl.dto.FilterTypeDTO;
+import gov.healthit.chpl.entity.FilterEntity;
+import gov.healthit.chpl.entity.FilterTypeEntity;
+import gov.healthit.chpl.exception.EntityRetrievalException;
+
+@Repository("filterDAO")
+public class FilterDAOImpl extends BaseDAOImpl implements FilterDAO {
+
+    @Override
+    public FilterDTO update(final FilterDTO dto) throws EntityRetrievalException {
+        FilterEntity entity = new FilterEntity();
+        entity.setFilter(dto.getFilter());
+        entity.setLastModifiedUser(Util.getAuditId());
+        entity.setFilterType(new FilterTypeEntity());
+        entity.getFilterType().setId(dto.getFilterType().getId());
+        entity.getFilterType().setName(dto.getFilterType().getName());
+        entity.setUser(getUserEntityById(dto.getUser().getId()));
```

Comment:
You shouldn't really have to set the UserEntity here. The entity could be set up so that it has a field that maps to the user_id column and an additional field, marked insertable=false and updatable=false that lazily fetches the UserEntity. The way SurveillanceEntity maps certifiedProductId and certifiedProduct fields is a good example. If the FilterEntity is set up that way then you only have to set the userId field here and shouldn't need the getUserEntityById method at all. I'm not 100% sure on this but I think the way it is currently might end up updating the row for this user in the user table (although would not change anything)?